### ISSUE TITLE
SOZI crash on launching

### ISSUE THREAD

**Issue by RafaelLinux** [sozi-projects/Sozi#603]
### Summary of your problem or feature request
Crash when launching

Anyway I execute sozi (from uncompressed directory or RPM installed) it ends immediately crashing:
```
[10876:0209/145028.132675:FATAL:gpu_data_manager_impl_private.cc(439)] GPU process isn't usable. Goodbye.
`trap' para punto de parada/seguimiento (`core' generado)
```
 
### Version of Sozi and other relevant software
* Which version of Sozi are you using?  Version 22.02.03

* Which drawing software did you use? (examples: Inkscape 0.48, LibreOffice Draw 4.2)  Inkscape 1.0.1 (3bc2e81, 2020-09-07)

* On which platform? openSUSE Tumbleweed

### Steps to reproduce the problem
1- Extract sozi or install from RPM
2- Launch sozi

What did you do to make the problem happen? Trying to execute sozi

Give enough details to enable us to reproduce the problem.
Using NVidia private drivers

### Expected behavior
What results did you expect? To launch sozi!!!

### Observed behavior
What results did you actually get? An empty windows appears and vanish

**Comment by edmundlaugasson** [comment]
Same issue also here. Tried AppImage version:
```
./Sozi-22.2.20-1645393457.AppImage 
[64702:0221/195253.125881:FATAL:gpu_data_manager_impl_private.cc(439)] GPU process isn't usable. Goodbye.
Trace/breakpoint trap (core dumped)
```
 also [Arch Linux version](https://aur.archlinux.org/packages/sozi):
```
sozi 
[64110:0221/195038.871795:FATAL:gpu_data_manager_impl_private.cc(439)] GPU process isn't usable. Goodbye.
Trace/breakpoint trap (core dumped)
```
Most concerning, that no way to start Sozi...

**Comment by aumouvantsillage** [comment]
> I hope developers may conclude, that more recent versions may be on wrong direction. Hopefully Sozi will be fixed!

Sozi is just a web application that runs in a special-purpose web browser via [electron](https://www.electronjs.org/).
What you observe is not the result of a change in Sozi itself, but probably a compatibility problem between recent versions of electron and your operating system.

Si I don't believe Sozi is developed in the "wrong direction". I don't see how the changes made in Sozi since version 19.04 could relate to this issue, except an upgrade of electron.

In my tests, Sozi 22.02.20 ran correctly in Debian 11, Ubuntu 20.04, Fedora 35 and Windows 10.
I have not tried openSUSE or ArchLinux yet but I will do that as soon as possible.

**Comment by edmundlaugasson** [comment]
My bad - certainly I don't know, what's under the hood. Just hard to determine, what can I do to get Sozi working. If there are people, who's work rely on Sozi, then it is quite challenging, if suddenly it does not work.

**Comment by aumouvantsillage** [comment]
> As there are people, who's work rely on Sozi, then it is quite challenging, if suddenly it does not work.

I can totally understand that, and I usually try to do my best to fix critical issues quickly.
But I cannot fix issues that I am not aware of :)

Based on your tests, it seems that Sozi has had launching issues since may 2020 (the publication date of Sozi 20.05), but as far as I remember, nobody reported the issue before now.

**Comment by edmundlaugasson** [comment]
Currently installed electron versions:
- electron 16.0.8-2
- electron13 13.6.9-1
- electron14 14.2.6-1
- electron15 15.3.7-2

I haven't installed them. These have been installed as dependencies by other programs. Do not know, which version will be used, when Sozi is launching or how to define, which version should be used. Interesting, that old 19.04.22 of Sozi version worked with these new electron versions.
I know, that electron is famous and perhaps good framework to use. As we see, quite complicated framework as easily can it be broken. If there could be possible to use some kind of more stable framework (or whatever you call it), it would be very much appreciated.

**Comment by aumouvantsillage** [comment]
> Currently installed electron versions

The current official Sozi package is based on electron 9.4.4.
Normally, electron comes bundled with Sozi, but I think that the ArchLinux User Repository provides it separately.

A quick look at the history shows the following evolution:

Sozi version | Electron version
---|---
19.04 | 3.02
20.05 | 6.1.4
21.02 | 9.2.1
21.08 and later | 9.4.4

> If there could be possible to use some kind of more stable, independent framework (or whatever you call it), it would be very much appreciated.

I have already considered moving away from Electron for various reasons.
There are alternatives, but they also have their shortcomings.

At this point, I will investigate this specific issue with Electron 9.4.4.
There are chances that it has already been reported for other Electron-based applications, and hopefully fixed in a more recent version of Electron.

**Comment by aumouvantsillage** [comment]
For reference, here is a [discussion about a similar issue for several Electron application in OpenSUSE Tumbleweed](https://forums.opensuse.org/showthread.php/560226-Electron-apps-fail-to-launch-FATAL-gpu_data_manager_impl_private-cc(415)-GPU-process-isn-t-usable).

**Comment by edmundlaugasson** [comment]
Thank you for comprehensive feedback! I mentioned current discussion also in Arch User Repository (AUR) close to Sozi package. Let's hope, that if Electron is not bundled there, it will be.
If Electron is bundled with Sozi, then any version of Sozi should run regardless of installed Electron version. Still there might some conflict, that is hard to determine - this could be, why still Sozi won't start in Arch. Just trying to figure out, what keeps back Sozi, that it won't start - even I downloaded directly from Sozi GitHub (where Electron should be bundled), unpacked and started manually from CLI.

> I have already considered moving away from Electron for various reasons.
> There are alternatives, but they also have their shortcomings.

You know better. I hope you will find a way, that do not overwhelm you and has less issues than currently with Electron.
Just quick look at https://alternativeto.net/software/electron/?license=opensource&platform=linux shows 23 alternatives to Electron (some of them may be discontinued) - hopefully there is at least one, that is more suitable than Electron. [Looks like](https://www.electronjs.org/releases/stable) even Electron stable is developed quite rapidly - hard to keep pace with it.

**Comment by aumouvantsillage** [comment]
I have successfully reproduced the issue in OpenSUSE Tumbleweed running in a virtual machine.

Commit 8c32c22 activates an option of Electron that fixes the issue for me in OpenSUSE.
With this fix, the AppImage also launches correctly in Manjaro, so I would expect it to run in ArchLinux as well.

You will find updated packages at https://github.com/sozi-projects/Sozi/releases/tag/v22.02.23-beta
At the moment, I am still tweaking the new package build process, so I don't provide *tar* bundles for Linux.

**Comment by aumouvantsillage** [comment]
@edmundlaugasson I have added an ArchLinux package (with extension `.pacman`) to release 22.02.23. It installs and runs correctly in a Manjaro VM.

**Comment by edmundlaugasson** [comment]
Thank you! This version works also in Arch Linux. Although using [EndeavourOS](https://endeavouros.com/), if to be precise but it is very close to Arch itself.
You know better, whether [such disabling](https://github.com/sozi-projects/Sozi/commit/8c32c2214dce4f07398497143b07ccf0334fb310) does affect graphical perfomance and/or appearance and on which extent, e.g. correct color rendering. Just wondering, why such issue suddenly appeared at all.

**Comment by aumouvantsillage** [comment]
> You know better, whether [such disabling](https://github.com/sozi-projects/Sozi/commit/8c32c2214dce4f07398497143b07ccf0334fb310) does affect graphical performance and/or appearance

My understanding is that it only disables [a security feature](https://www.electronjs.org/docs/latest/tutorial/sandbox/). It should not affect performance or appearance.

I don't have enough expertise in the architecture of Electron to measure the risk when disabling the sandbox. And since some distributions can run Electron applications with the sandbox enabled, it is possible that a safer solution exists.

**Comment by RafaelLinux** [comment]
It seems it is solved in test version Sozi-22.2.23-1645647480, no crash.

**Comment by edmundlaugasson** [comment]
Sounds like this works also with current version:
`sozi --disable-gpu-sandbox`
Sounds like [Chromium CLI parameter](https://peter.sh/experiments/chromium-command-line-switches/).

**Comment by edmundlaugasson** [comment]
Still pre-release and no release? Reported issue still persist with current release but closed without any commit or release? Almost a month passed since the pre-release.

**Comment by aumouvantsillage** [comment]
> Reported issue still persist with current release but issue closed without any commit or release?

Let me tell you that I find your comment a little unfair.

With GitHub, there is an option to close an issue as soon as a fix is committed. Many projects use it, including Sozi, which results in issues being closed even if no release has been published.
I could have used this option in commit 8c32c22 but I chose not to. I published a pre-release and I waited for feedback before closing the issue.

After one month without additional info, I consider that this issue can be closed. Without any other bug report specific to the latest pre-release, I will also promote it to an official release.

**Comment by RafaelLinux** [comment]
For my part, and as I am "guilty" for not having reported it earlier, I summarise the state of the problem in my equipment as of today:
Version **previous** to  Sozi-22.2.23-1645647480 crash, regardless of whether it is RPM or AppImage version.
I must add that install Sozi-22.2.23-1645647480  RPM version give a warning about NSS file is missing.

**Comment by edmundlaugasson** [comment]
As I mentioned already 24 days ago, in my distro the [Sozi-22.2.23-1645647480.AppImage](https://github.com/sozi-projects/Sozi/releases/download/v22.02.23-beta/Sozi-22.2.23-1645647480.AppImage) works. Used for quick testing. I guess pacman version will also as I have that package manager. But I don't want manually install packages as Sozi is available also via package manager.
If the wait period is one month then it is understandable, although quite long to wait new, fixed version.

**Comment by aumouvantsillage** [comment]
> I will try to reopen the issue, as edmundlaugasson is right: the stable version still doesn't work.

As I tried to explain in my previous comment, closing issues and releasing a stable version are separate processes.
The fact that an issue is closed does not always mean that the fix is published in a stable release.
Different projects have different rules for managing issues and releases.

My usual process is:

1. Push a fix to the source repository.
2. Mark the issue with the label "Done".
3. Publish a pre-release.
4. Close the issue if the fix is confirmed.
5. Repeat steps 1 to 4 for another issue.
6. Publish a stable release.

But it is not a "formal" process.
Sometimes, I close an issue immediately in step 1.
Often, users decide to close their own issues after step 3.
Because for many people, there is no reason to keep an issue open when a fix has been published, even as a pre-release.

> If the wait period is one month then it is understandable, although quite long to wait new, fixed version.

Please remember that Sozi is a side project that I develop mostly alone in my spare time.
I have a job and a family, so I work on Sozi when I can and when I want.
I have no obligation to fix issues, provide free support, or comply with a specific process or schedule.
So I think that you should adjust your expectations to what I can provide, and not the other way round.

**Comment by edmundlaugasson** [comment]
I just would like to say, that if there is no issues, then probably it is good time to make a release from pre-release.
I know all this (you repeatedly remind this) and nobody don't want irritate you. It is just regular communication and no hard feelings.

**Comment by aumouvantsillage** [comment]
> I know all this (you repeatedly remind this) and nobody don't want irritate you. It is just regular communication and no hard feelings.

OK. It is not always easy to tell whether a message is a question, a factual statement or a complaint.

Since [I proposed to promote the current pre-release as the new stable release](https://github.com/sozi-projects/Sozi/issues/603#issuecomment-1073646334), the purpose of the next two messages by @RafaelLinux and you was not clear to me.

**Comment by aumouvantsillage** [comment]
FYI, I have published Sozi 22.03.25 as the new official release.

**Comment by RafaelLinux** [comment]
> FYI, I have published Sozi 22.03.25 as the new official release.

Yes, it works (I tried AppImage version, not RPM one).
Thank for all your work!!!
